### PR TITLE
Add tankerkoenig integration

### PR DESCRIPTION
**Pull request by guillempages**
**Description:**
Add a new integration for tankerkoenig.de gas station prices.

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#28661
## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Comment by guillempages** [comment]
@frenck I updated the PR after your review, since I added a new optional configuration option in the parent PR. (It was requested by a reviewer and it made sense to me).
I'm not sure what the process is here (this is my first integration); do you need to re-approve, or will you do that once the parent PR gets merged?

**Comment by frenck** [comment]
The PR is fine at this point and is awaiting the parent PR to get merged. Don't worry about the failed `/` build. That is a pending change out of your control.

**Review by frenck — APPROVED** [review]

**Review by frenck — APPROVED** [review]

**Pull request merged**
